Tidy up some of the ST I2C/SPI/SSC Kconfig options, so that SSC selection
is now automatic. Also removed references to explict chips, which are
difficult to maintain, and shouldn't be needed.

Signed-off-by: Stuart Menefy <stuart.menefy@st.com>
Index: linux/drivers/i2c/busses/Kconfig
===================================================================
--- linux.orig/drivers/i2c/busses/Kconfig	2007-07-16 15:58:11.250283000 +0100
+++ linux/drivers/i2c/busses/Kconfig	2007-07-16 15:58:18.060323000 +0100
@@ -529,37 +529,34 @@
 
 choice
 	prompt "STMicroelectronics I2C support"
-	depends on I2C && CPU_SUBTYPE_STB7100
-
+	depends on I2C && CPU_SUBTYPE_ST40
 	help
-	  STMicroelectronics SoCs have I2C drivers exist for bit-banging using PIO pins or
-          running an I2C bus using the hardware SSC
+	  STMicroelectronics SoCs support the I2C bus using either a bit-banging
+	  PIO driver or using the hardware SSC.
 
-	config I2C_ST40_PIO
+config I2C_ST40_PIO
 	tristate "I2C bit-banging PIO driver"
 	select I2C_ALGOBIT
 	help
-
 	  If you say yes to this option, support will be included for the
-	  PIO based I2C on the STm8000, STi5528 and STb7100.
+	  PIO driven I2C in bit-banging mode.
 
 	  This driver can also be built as a module.  If so, the module
 	  will be called i2c-st40-pio
 
-	config I2C_STM
+config I2C_STM
 	tristate "I2C hardware SSC driver"
-       	select STM_SSC
+	select STM_SSC
 	help
-
 	  If you say yes to this option, support will be included for the
-	  I2C-SSC for the STM SOCs: STi5528, STm8000 and the STb7100
+	  SSC driven I2C.
+
+endchoice
 
 config I2C_STM_HW_FIFO
 	bool "I2C hardware SSC Fifo support"
 	depends on I2C_STM
 	help
-	Enable this option to use the ssc hardware fifo support
-
-endchoice
+	  Enable this option to print out information on STM I2C (SSC) device driver
 
 endmenu
Index: linux/drivers/stm/Kconfig
===================================================================
--- linux.orig/drivers/stm/Kconfig	2007-07-16 15:58:11.271281000 +0100
+++ linux/drivers/stm/Kconfig	2007-07-16 15:58:18.065323000 +0100
@@ -53,7 +53,6 @@
 
 config STM_SSC
        bool "STM Synchronous Serial Controller"
-       depends on !I2C_ST40_PIO
        default y
        ---help---
          This driver is required to support the I2C and the SPI
@@ -63,14 +62,14 @@
 	bool "STM SSC Debug"
 	depends on STM_SSC
 	default n
-        ---help---
-        Enable this option to print out information on STM SSC device driver
+	---help---
+	  Enable this option to print out information on STM SSC device driver
 
 config STM_SPI
 	bool "STM Serial Peripheral Interface"
-	depends on STM_SSC
+	select STM_SSC
 	---help---
-	This driver is required to support the SPI Protocol
+	  This driver is required to support the SPI Protocol
 
 config  STM_SPI_DEBUG
         bool "STM SPI Debug"
Index: linux/include/linux/st_soc.h
===================================================================
--- linux.orig/include/linux/st_soc.h	2007-07-16 15:58:16.192312000 +0100
+++ linux/include/linux/st_soc.h	2007-07-16 15:58:18.078324000 +0100
@@ -77,6 +77,9 @@
         unsigned char sdinpin;
 };
 
+#define SSC_I2C_CAPABILITY  0x1
+#define SSC_SPI_CAPABILITY  0x2
+
 struct plat_ssc_data {
         unsigned short  capability;   /* bitmask on the ssc capability */
         struct plat_ssc_pio_t *pio;   /* the Pio map */
